client/src/Frontend/Components/Icons/Icons.test.ts

```diff
@@ -0,0 +1,48 @@
+import { cleanup, render, screen } from '@testing-library/svelte';
+import { afterEach, expect } from 'vitest';
+import FoundRuins from './FoundRuins.svelte';
+import Quasar from './Quasar.svelte';
+
+afterEach(() => cleanup());
+
+test('Quasar icon: default', () => {
```

Comment:
@phated is there a way to test these more generically? ie, I just pass in the name of the icon and the it does the three following tests?

In that same vein, I noticed that `Icons.tsx` has a bunch of the same code that just differs in name, image source, and alt text. Is there any way to reduce boilerplate here or better to hardcode. 

As an example, I'm thinking like:
```ts
function buildIcon(name: string, imgPath: string): JSX.element
```
and we'd just pass in the name and url of the icon. lmk!

Reply:
> is there a way to test these more generically? ie, I just pass in the name of the icon and the it does the three following tests?

Since the tests are just TS, you could easily write a helper function to produce a "suite":
```ts
function suite(iconName: string) {
   // ...test stuff here but use iconName where you need it!
}
```

> In that same vein, I noticed that Icons.tsx has a bunch of the same code that just differs in name, image source, and alt text. Is there any way to reduce boilerplate here or better to hardcode.

I think that is a great idea and I even did it for a lot of the other icons in the project as part of the `ui` package. I'm torn on how much "improvement" to do vs refactor during the changes. 🤔

Reply:
I agree we don't want to refactor too much. But the more we refactor UI, the fewer components we have to port to Svelte 🤷